pkg/reconciler/buildrun/resources/results.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package resources
+
+import (
+	"fmt"
+
+	buildv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+type sourceResult struct {
+	defaultSource,
+	bundleSource buildv1alpha1.SourceResult
+}
+
+const (
+	defaultSourceName       = "default"
+	commitSHAResult         = "commit-sha"
+	commitAuthorResult      = "commit-author"
+	bundleImageDigestResult = "bundle-image-digest"
+	imageDigestResult       = "image-digest"
+	imageSizeResult         = "image-size"
+)
+
+// UpdateBuildRunUsingTaskResults surface the task results
+// to the buildrun
+func UpdateBuildRunUsingTaskResults(
+	buildRun *buildv1alpha1.BuildRun,
+	lastTaskRun *v1beta1.TaskRun,
+) {
+	var sources sourceResult
+
+	// Initializing source result
+	sources.defaultSource.Git = &buildv1alpha1.GitSourceResult{}
+	sources.bundleSource.Bundle = &buildv1alpha1.BundleSourceResult{}
```

Comment:
Could we move that into the `if` blocks that follow? Theoretically speaking, we do not need to initialize both as usually only one of them will be used.

Reply:
I have moved the initialization to the `switch` case of `func updateBuildRunStatus()`, so that it will only initialize the source being used.